Datasnap/Classes/Utilities/Datasnap.m

```diff
@@ -211,41 +223,26 @@ - (void)checkQueue
     }
 }
 
-- (void)flushEvents
+- (NSString*)sha1:(NSString*)input
 {
-    [self.eventQueue flushQueue];
-}
+    const char* cstr = [input cStringUsingEncoding:NSUTF8StringEncoding];
+    NSData* data = [NSData dataWithBytes:cstr length:input.length];
 
-- (void)genericEvent:(NSMutableDictionary*)eventDetails
-{
-    [self eventHandler:eventDetails];
-}
+    uint8_t digest[CC_SHA1_DIGEST_LENGTH];
 
-- (void)eventHandler:(NSMutableDictionary*)eventDetails
-{
-    NSMutableDictionary* eventDetailsCopy = [eventDetails mutableCopy];
-    NSDictionary* deviceInfo = [self deviceInfoDictionary];
-    [eventDetailsCopy addEntriesFromDictionary:@{ @"organization_ids" : @[ self.organizationId ],
-        @"project_ids" : @[ self.projectId ],
-        @"sdk_version" : @"1.0.4",
-        @"device_info" : @{
-            @"platform" : [deviceInfo objectForKey:@"platform"],
-            @"system_name" : [deviceInfo objectForKey:@"system_name"],
-            @"system_version" : [deviceInfo objectForKey:@"system_version"]
-        }
-    }];
-    [self.eventQueue recordEvent:eventDetailsCopy];
-    [self checkQueue];
+    CC_SHA1(data.bytes, data.length, digest);
+
+    NSMutableString* output = [NSMutableString stringWithCapacity:CC_SHA1_DIGEST_LENGTH * 2];
+
+    for (int i = 0; i < CC_SHA1_DIGEST_LENGTH; i++)
+        [output appendFormat:@"%02x", digest[i]];
+
+    return output;
 }
 
-- (NSDictionary*)deviceInfoDictionary
+- (void)flushEvents
```

Comment:
Unless it's for personal readability preference, I feel like flushEvents isn't necessary since eventQueue property is exposed anyways.